File: backend/src/main/java/csp/inventory/Portfolio.java
package csp.inventory;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity
public class Portfolio {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER, orphanRemoval = true)
    @JoinColumn(name = "portfolio_id")
    private List<Item> itemList = new ArrayList<>();
    private BigDecimal currentValue;
    private BigDecimal totalPurchasePrice;
    private BigDecimal changePercentage;
}

File: frontend/src/types/portfolio.ts
export type Portfolio = {
  id: number;
  itemList: Item[];
  currentValue: number;
  totalPurchasePrice: number;
  changePercentage: number;
}

export type Item = {
  id: number;
  name: string;
  quantity: number;
  currentPrice: number;
  purchasePrice: number;
}
